ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController.swift

```diff
@@ -3240,7 +3240,9 @@ extension BrowserViewController {
   }
 
   func openBraveLeo(with query: String? = nil) {
-    if !FeatureList.kAIChat.enabled {
+    if !FeatureList.kAIChat.enabled
+      || !AIChatUtils.isAIChatEnabled(for: profileController.profile.prefs)
```

Comment:
May be able to remove the feature list check here and rely only on `AIChatUtils.isAIChatEnabled` which under the hood will also check the feature flag